pkg/platform/packet/packet.go

```diff
@@ -294,3 +330,117 @@ func (c *config) checkWorkerPoolNamesUnique() hcl.Diagnostics {
 
 	return diagnostics
 }
+
+func (c *config) checkReservationIDs() hcl.Diagnostics {
+	var diagnostics hcl.Diagnostics
+
+	d := checkEachReservation(c.ReservationIDs, c.ReservationIDsDefault, "controller", c.ClusterName)
+	diagnostics = append(diagnostics, d...)
+
+	for _, w := range c.WorkerPools {
+		d := checkEachReservation(w.ReservationIDs, w.ReservationIDsDefault, "worker", w.Name)
+		diagnostics = append(diagnostics, d...)
+	}
+
+	return diagnostics
+}
+
+// checkEachReservation checks that hardware reservations are in the correct
+// format and, when it will cause problems, that reservation IDs values in this
+// pool are not mixed between using "next-available" and specific UUIDs, as this
+// can't work reliably.
+// For more info, see comment when calling terraformCreateReservations()
+func checkEachReservation(reservationIDs map[string]string, resDefault, nodeRole, name string) hcl.Diagnostics {
```

Comment:
Ditto here, it would be nice to have some tests for this function.

Reply:
idem